core/server/policy_violations.go

```diff
@@ -0,0 +1,267 @@
+	"google.golang.org/protobuf/types/known/anypb"
+	"google.golang.org/protobuf/types/known/wrapperspb"
+	v1 "k8s.io/api/core/v1"
+	k8sFields "k8s.io/apimachinery/pkg/fields"
+	k8sLabels "k8s.io/apimachinery/pkg/labels"
+	sigsClient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type validationList struct {
+	Validations []*pb.PolicyValidation
+	Token       string
+	Errors      []*pb.ListError
+}
+
+func (cs *coreServer) ListPolicyValidations(ctx context.Context, m *pb.ListPolicyValidationsRequest) (*pb.ListPolicyValidationsResponse, error) {
+	var respErrors []*pb.ListError
+	clustersClient, err := cs.clustersManager.GetImpersonatedClient(ctx, auth.Principal(ctx))
+	if err != nil {
+		if merr, ok := err.(*multierror.Error); ok {
+			for _, err := range merr.Errors {
+				if cerr, ok := err.(*clustersmngr.ClientError); ok {
+					respErrors = append(respErrors, &pb.ListError{ClusterName: cerr.ClusterName, Message: cerr.Error()})
+				}
+			}
+		}
+	}
+	labelSelector, err := k8sLabels.ValidatedSelectorFromSet(map[string]string{
+		"pac.weave.works/type": "Admission"})
+	if err != nil {
+		return nil, fmt.Errorf("error building selector for events query: %v", err)
+	}
+
+	fieldSelectorSet := map[string]string{
+		"type": "Warning",
+	}
+
+	if m.Application != "" {
+		fieldSelectorSet["involvedObject.name"] = m.Application
+		fieldSelectorSet["involvedObject.kind"] = m.Kind
+	}
+
+	if m.Namespace != "" {
+		fieldSelectorSet["involvedObject.namespace"] = m.Namespace
+	}
+
+	if m.PolicyId != "" {
+		fieldSelectorSet["involvedObject.fieldPath"] = m.PolicyId
+	}
+
+	fieldSelector := k8sFields.SelectorFromSet(fieldSelectorSet)
+
+	opts := []sigsClient.ListOption{}
+	if m.Pagination != nil {
+		opts = append(opts, sigsClient.Limit(m.Pagination.PageSize))
+		opts = append(opts, sigsClient.Continue(m.Pagination.PageToken))
+	}
+	opts = append(opts, &sigsClient.ListOptions{
+		LabelSelector: labelSelector,
+		FieldSelector: fieldSelector,
+	})
+	opts = append(opts, sigsClient.InNamespace(v1.NamespaceAll))
+
+	validationsList, err := cs.listEvents(ctx, clustersClient, m.ClusterName, false, opts)
+	if err != nil {
+		return nil, fmt.Errorf("error getting events: %v", err)
+	}
+	respErrors = append(respErrors, validationsList.Errors...)
+	policyviolationlist := pb.ListPolicyValidationsResponse{
+		Total:         int32(len(validationsList.Validations)),
+		Violations:    validationsList.Validations,
+		Errors:        respErrors,
+		NextPageToken: validationsList.Token,
+	}
+	return &policyviolationlist, nil
+}
+
+func (cs *coreServer) GetPolicyValidation(ctx context.Context, m *pb.GetPolicyValidationRequest) (*pb.GetPolicyValidationResponse, error) {
+	clusterClient, err := cs.clustersManager.GetImpersonatedClient(ctx, auth.Principal(ctx))
+	if err != nil {
+		return nil, fmt.Errorf("error getting impersonating client: %w", err)
+	}
+
+	selector, err := k8sLabels.ValidatedSelectorFromSet(map[string]string{
+		"pac.weave.works/type": "Admission",
```

Comment:
can we have this as part of the request payload so we can easily use the API for audit too